### PR TITLE
Added logger class

### PR DESCRIPTION
**Pull request by Hipst0man**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Summary of change

ToDo add parser to get configuration for logger

### CODE REVIEW

File: CPPLogger/CPPLogger.h [hunk 1]
```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include <iostream>
+#include <string>
+
+#include "P7_Client.h"
+#include "P7_Trace.h"
+
+
+enum LogLevels
```
Comment (CharlesLyleLaRue) [inline CPPLogger/CPPLogger.h:10]: It's better to use enum class instead of enum.

File: CPPLogger/CPPLogger.h [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+class ILogger
+{
+public:
+	virtual ILogger& operator() (const std::string& func_name, const std::string& file_name, short level) = 0;
+	virtual ILogger& operator<< (const char* log_message) = 0;
+	virtual void set_filter_level(LogLevels level) = 0;
+};
+
+class Logger : public ILogger
```
Comment (CharlesLyleLaRue) [inline CPPLogger/CPPLogger.h]: If you will not inherit from the class "Logger" put keyword "final".
Like this:
class Logger final : public ILogger

File: CPPLogger/CPPLogger.cpp [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+// PVS-Studio Static Code Analyzer for C, C++, C#, and Java: http://www.viva64.com
+
+#include "pch.h"
+#include "framework.h"
+
+#include "CPPLogger.h"
+
+const wchar_t* CLIENT_INIT_PARAM = L"/P7.Sink=FileTxt /P7.Dir=C:\Program Files\Logs\ /P7.Format=\"%ti - %tf [%lv] - [%fs] [%fn] %ms\"";
+const wchar_t* TRACE_CHANNEL = L"Trace";
+const bool TRACE_ID = NULL;
```
Comment (CharlesLyleLaRue) [inline CPPLogger/CPPLogger.cpp]: Why did you initiate const bool as NULL?

File: CPPLogger/CPPLogger.cpp [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+	m_trace = P7_Create_Trace(m_client, TRACE_CHANNEL);
+}
+
+Logger::~Logger()
+{
+	m_trace->Release();
+	m_client->Release();
+}
+
+Logger& Logger::operator()(const std::string& func, const std::string& filename, short level)
```
Comment (CharlesLyleLaRue) [inline CPPLogger/CPPLogger.cpp:28]: Maybe check input parameters of this method.

**Review by CharlesLyleLaRue — CHANGES_REQUESTED** [review]
No chiki-bomboni for you today =)
Please, check my comments.

File: CPPLogger/CPPLogger.h (hunk 1 above)
Comment (Hipst0man) [reply CPPLogger/CPPLogger.h:10]: I need to cast from my enum to lib enum inside one of my func. If i use enum class i couldn't do it.

**Review by KostiantynYehorov — DISMISSED** [review]
All right

**Review by CharlesLyleLaRue — DISMISSED** [review]
LGTM

**Review by CharlesLyleLaRue — APPROVED** [review]
LGTM

**Review by OksanaBisyk-TomTom — APPROVED** [review]

**Review by KostiantynYehorov — APPROVED** [review]

**Pull request merged**